### PR TITLE
85 Add support for skipClient option when generating a JHipster app

### PR DESCRIPTION
**Pull request by joewhite101**

### CODE REVIEW

**Comment by joewhite101** [comment]
Fixes https://github.com/jhipster/jhipster-quarkus/issues/85

**Review by nshaw — APPROVED** [review]
Looks good to me. I was able to run mvnw successfully.

File: generators/server/templates/quarkus/pom.xml.ejs [hunk 1]
```diff
@@ -451,58 +453,61 @@
...
-                        </executions>
-                    </plugin>
-                </plugins>
-            </build>
-            <properties>
-                <!-- default Quarkus profiles -->
-                <quarkus.profile>dev</quarkus.profile>
-            </properties>
-        </profile>
+
```
Comment (avdev4j) [inline generators/server/templates/quarkus/pom.xml.ejs:456]: Remove empty line

File: generators/server/templates/quarkus/pom.xml.ejs [hunk 2]
```diff
@@ -451,58 +453,61 @@
...
+                                    <goals>
+                                        <goal>npm</goal>
+                                    </goals>
+                                    <phase>generate-resources</phase>
+                                    <configuration>
+                                        <arguments>run webpack:build</arguments>
+                                        <environmentVariables>
+                                            <APP_VERSION>${project.version}</APP_VERSION>
+                                        </environmentVariables>
+                                        <npmInheritsProxyConfigFromMaven>false</npmInheritsProxyConfigFromMaven>
```
Comment (avdev4j) [inline generators/server/templates/quarkus/pom.xml.ejs:498]: Remove empty line

File: generators/server/templates/quarkus/pom.xml.ejs [hunk 3]
```diff
@@ -544,6 +549,8 @@
                             </filesets>
                         </configuration>
                     </plugin>
+
```
Comment (avdev4j) [inline generators/server/templates/quarkus/pom.xml.ejs:552]: Remove empty line

File: generators/server/templates/quarkus/pom.xml.ejs [hunk 4]
```diff
@@ -594,6 +601,8 @@
                             </execution>
                         </executions>
                     </plugin>
+                    <%_ } _%>
+
```
Comment (avdev4j) [inline generators/server/templates/quarkus/pom.xml.ejs:605]: Remove empty line

File: generators/server/templates/quarkus/pom.xml.ejs [hunk 5]
```diff
@@ -451,58 +453,61 @@
...
-                </plugins>
-            </build>
-            <properties>
-                <!-- default Quarkus profiles -->
-                <quarkus.profile>dev</quarkus.profile>
-            </properties>
-        </profile>
+
+        <%_ if (!skipClient) { _%>
+            <profile>
```
Comment (avdev4j) [inline generators/server/templates/quarkus/pom.xml.ejs:458]: It seems that the indentation is not the right one. It should be 4 spaces left below `<profiles>`.

File: generators/server/files.js [hunk 6]
```diff
@@ -299,10 +299,7 @@ const serverFiles = {
                     file: 'package/web/rest/AccountResource.java',
                     renameTo: generator => `${generator.javaDir}web/rest/AccountResource.java`
                 },
-                {
-                    file: 'package/web/rest/SpaFilter.java',
-                    renameTo: generator => `${generator.javaDir}web/rest/SpaFilter.java`
-                },
+
```
Comment (avdev4j) [inline generators/server/files.js:302]: Remove empty line

File: generators/server/files.js [hunk 7]
```diff
@@ -321,6 +318,16 @@ const serverFiles = {
                 }
             ]
         },
+        {
+            path: SERVER_MAIN_SRC_DIR,
+            templates: [
+                {
+                    condition: generator => !generator.skipClient,
+                    file: 'package/web/rest/SpaFilter.java',
```
Comment (avdev4j) [inline generators/server/files.js:326]: We should add a test case to ensure that the file is not generate when we choose no client framework.

**Review by avdev4j — CHANGES_REQUESTED** [review]

**Comment by joewhite101** [comment]
Not ready to merge!

@avdev4j  Updated formatting based on notes.

Having an issue with the unit test. For some reason I can't get `skipClient` to pass through to the generator as "false". It is always set to true in the tests. Works correctly when running manually.  Tried using the app generator as the base test but the same issue.

**Comment by joewhite101** [comment]
And my local results are the opposite of git ...

**Comment by avdev4j** [comment]
> Having an issue with the unit test. For some reason I can't get `skipClient` to pass through to the generator as "false". It is always set to true in the tests. Works correctly when running manually. Tried using the app generator as the base test but the same issue.

Wait! You remind me something, I think I already met this problem. The skipClient flag is ignore when we test it through blueprint tests. I don't why. Maybe we should open a ticket on the main repo. It's tricky but we need to fix that.

**Comment by joewhite101** [comment]
> > Having an issue with the unit test. For some reason I can't get `skipClient` to pass through to the generator as "false". It is always set to true in the tests. Works correctly when running manually. Tried using the app generator as the base test but the same issue.
> 
> Wait! You remind me something, I think I already met this problem. The skipClient flag is ignore when we test it through blueprint tests. I don't why. Maybe we should open a ticket on the main repo. It's tricky but we need to fix that.

You probably know more than I do. Do you want to open that ticket or should I?

**Comment by avdev4j** [comment]
I can, If you want. I did not do it in the past (my bad) I can fix that ;)

**Comment by joewhite101** [comment]
@avdev4j Sounds good. Let me know the number. I think that one will need a bug bounty. That will be an issue for my other blueprint as well

**Comment by avdev4j** [comment]
@joewhite101 I know why the skip-client doesn't work when testing server subgen (and blueprint).

This option is init by the app subgen and not by the server. If we want to test that, we need to switch the test from server to app and pass the right flag value.

I'll do a commit to fix that.

**Comment by avdev4j** [comment]
the bug has been posted
https://github.com/jhipster/generator-jhipster/issues/13180

**Comment by avdev4j** [comment]
I have intentionally remove the test related to the SpaFilter because we can't test it properly for the moment. let's merge this one and wait for a fix to test this specific scenario.

**Review by avdev4j — APPROVED** [review]

**Pull request merged**